test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,163 @@
+    testHelper(msg = "test")(_ => reporter.print(NoPosition, "test", reporter.INFO))
+    testHelper(msg = "test", severity = "warning: ")(_ => reporter.print(NoPosition, "test", reporter.WARNING))
+    testHelper(msg = "test", severity = "error: ")(_ => reporter.print(NoPosition, "test", reporter.ERROR))
+    testHelper(posWithSource, msg = "test")(_ => reporter.print(posWithSource, "test", reporter.INFO))
+    testHelper(posWithSource, msg = "test", severity = "warning: ")(_ => reporter.print(posWithSource, "test", reporter.WARNING))
+    testHelper(posWithSource, msg = "test", severity = "error: ")(_ => reporter.print(posWithSource, "test", reporter.ERROR))
+  }
+
+
+  @Test
+  def printColumnMarkerTest(): Unit = {
+    testHelper(msg = "")(_ => reporter.printColumnMarker(NoPosition))
+
+    reporter.printColumnMarker(posWithSource)
+    assertEquals("    ^", writerOut.toString.lines.next)
+    writerOut.reset
+  }
+
+
+  @Test
+  def displayTest(): Unit = {
+    testHelper(msg = "Testing display")(_ => reporter.display(NoPosition, "Testing display", reporter.INFO))
+    testHelper(msg = "Testing display", severity = "warning: ")(_ => reporter.display(NoPosition, "Testing display", reporter.WARNING))
+    testHelper(msg = "Testing display", severity = "error: ")(_ => reporter.display(NoPosition, "Testing display", reporter.ERROR))
+    testHelper(posWithSource, msg = "Testing display")(_ => reporter.display(posWithSource, "Testing display", reporter.INFO))
+    testHelper(posWithSource, msg = "Testing display", severity = "warning: ")(_ => reporter.display(posWithSource, "Testing display", reporter.WARNING))
+    testHelper(posWithSource, msg = "Testing display", severity = "error: ")(_ => reporter.display(posWithSource, "Testing display", reporter.ERROR))
+
+    reporter.ERROR.count += 1
+    testHelper(posWithSource, msg = "Testing display for maxerrs to pass", severity = "error: ")(_ => reporter.display(posWithSource, "Testing display for maxerrs to pass", reporter.ERROR))
+    reporter.ERROR.count += 1
+    testHelper(msg = "")(_ => reporter.display(posWithSource, "Testing display for maxerrs to fail", reporter.ERROR))
+
+    reporter.WARNING.count += 1
+    testHelper(posWithSource, msg = "Testing display for maxwarns to pass", severity = "warning: ")(_ => reporter.display(posWithSource, "Testing display for maxwarns to pass", reporter.WARNING))
+    reporter.WARNING.count += 1
+    testHelper(msg = "")(_ => reporter.display(posWithSource, "Testing display for maxwarns to fail", reporter.WARNING))
+  }
+
+
+  @Test
+  def finishTest(): Unit = {
+    reporter.resetCount(reporter.ERROR)
+    reporter.resetCount(reporter.WARNING)
+    testHelper(msg = "")(_ => reporter.finish())
+
+    reporter.ERROR.count = 10
+    reporter.WARNING.count = 3
+    reporter.finish()
+    val it = writerOut.toString.lines
+    assertEquals("three warnings found", it.next)
+    assertEquals("10 errors found", it.next)
+    writerOut.reset
+  }
```

Comment:
Should I use a separate `BAOS` for each test as well and pass it as an argument in `testHelper` or let it remain a global as it is now?